client/src/main/java/org/apache/rocketmq/client/impl/consumer/ConsumeMessageOrderlyService.java

```diff
@@ -92,10 +92,22 @@ public void run() {
         }
     }
 
-    public void shutdown() {
+    @Override
+    public void shutdown(long awaitTerminateMillis) {
         this.stopped = true;
         this.scheduledExecutorService.shutdown();
         this.consumeExecutor.shutdown();
+        //await to consume
+        if (awaitTerminateMillis > 0) {
+            try {
+                this.consumeExecutor.awaitTermination(awaitTerminateMillis,TimeUnit.MILLISECONDS);
+                if (!this.consumeExecutor.isTerminated()) log.info("There are messages still being consumed in thread pool, but not going to await them anymore. Have awaited for {} ms",awaitTerminateMillis);
```

Comment:
`shutdown` is the first choice, `shutdownNow` will be called if timeout.

And `develop` will be merged to `master` in next release, please refer to our new [branching model](http://rocketmq.apache.org/docs/branching-model).